core/node/events/stream.go

```diff
@@ -671,6 +672,54 @@ func (s *Stream) GetMiniblocks(
 		return nil, false, err
 	}
 
+	// if stream is external, we need to read from external storage
+	location, err := s.params.Storage.GetMediaStreamLocation(ctx, s.streamId)
```

Comment:
Why is the logic to fetch block data put here and not in the storage layer that makes how data is stored transparent for the rest of the node logic?

For snapshots the caller passes the `omitSnapshot` bool as indication if it wants snapshot data or not. For media streams I _think_ that ReadMiniblocks will always want to return block data?

Reply:
The storage layer only handle the psql specific operations (like read miniblocks), this requires access to the external media stream interface (download chunk from s3 or aws), which is outside of the storage layer. The choice was to not contaminate local storage logic with external API code. I guess the most elegant way would be to create another middleware layer to handle this, so it's transparent to this code here, and can call both local and remote storage functions.